shell/platform/linux/public/flutter_linux/fl_method_response.h

```diff
@@ -0,0 +1,187 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_METHOD_RESPONSE_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_METHOD_RESPONSE_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <glib-object.h>
+
+#include "fl_value.h"
+
+G_BEGIN_DECLS
+
+/**
+ * FlMethodResponseError:
+ * @FL_METHOD_RESPONSE_ERROR_FAILED: Call failed due to an unspecified error.
+ * @FL_METHOD_RESPONSE_ERROR_REMOTE_ERROR: An error was returned by the other
+ * side of the channel.
+ * @FL_METHOD_RESPONSE_ERROR_NOT_IMPLEMENTED: The requested method is not
+ * implemented.
+ *
+ * Errors for #FlMessageResponse objects to set on failures.
+ */
+#define FL_METHOD_RESPONSE_ERROR fl_method_response_error_quark()
+
+typedef enum {
+  FL_METHOD_RESPONSE_ERROR_FAILED,
+  FL_METHOD_RESPONSE_ERROR_REMOTE_ERROR,
+  FL_METHOD_RESPONSE_ERROR_NOT_IMPLEMENTED,
+} FlMethodResponseError;
+
+GQuark fl_method_response_error_quark(void) G_GNUC_CONST;
+
+G_DECLARE_DERIVABLE_TYPE(FlMethodResponse,
+                         fl_method_response,
+                         FL,
+                         METHOD_RESPONSE,
+                         GObject)
+
+struct _FlMethodResponseClass {
+  GObjectClass parent_class;
+};
+
+G_DECLARE_FINAL_TYPE(FlMethodSuccessResponse,
+                     fl_method_success_response,
+                     FL,
+                     METHOD_SUCCESS_RESPONSE,
+                     FlMethodResponse)
+
+G_DECLARE_FINAL_TYPE(FlMethodErrorResponse,
+                     fl_method_error_response,
+                     FL,
+                     METHOD_ERROR_RESPONSE,
+                     FlMethodResponse)
+
+G_DECLARE_FINAL_TYPE(FlMethodNotImplementedResponse,
+                     fl_method_not_implemented_response,
+                     FL,
+                     METHOD_NOT_IMPLEMENTED_RESPONSE,
+                     FlMethodResponse)
+
+/**
+ * FlMethodResponse:
+ *
+ * #FlMethodResponse contains the information returned when a #FlMethodChannel
+ * method call returns successfully.
+ */
+
+/**
+ * FlMethodSuccessResponse:
+ *
+ * #FlMethodError contains the information returned when a #FlMethodChannel
```

Comment:
I'm confused by this comment. Is it just copypasta on the type?

Reply:
It is...